### PR TITLE
Obj_Ice_Poly

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.h [hunk 1]
```diff
@@ -7,11 +7,18 @@ struct ObjIcePoly;
 
 typedef void (*ObjIcePolyActionFunc)(struct ObjIcePoly*, GlobalContext*);
 
+#define OBJICEPOLY_GET_FF00(thisx) (((thisx)->params >> 8) & 0xFF)
+
+#define OBJICEPOLY_FF_FF 0xFF
```
Comment (isghj5) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.h:12]: Right before this got decomped I noticed this shows up a lot with actors that have switch flags, where it checks the max value of 0x7F or 0xFF, and if thats the case the variant ignores switch flags, like this actor here.

Starting to think there should be a general define somewhere instead like "SWITCH_FLAG_IGNORE" or something

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 2]
```diff
@@ -32,37 +34,327 @@ const ActorInit Obj_Ice_Poly_InitVars = {
...
+    spAC.y = this->actor.scale.y;
+    spAC.z = 0.0f;
+
+    spA0.x = 0.0f;
+    spA0.y = this->actor.scale.y;
+    spA0.z = 0.0f;
+
+    for (i = 0; i < 2; i++) {
+        temp_f20 = Rand_ZeroOne();
+        if (Rand_ZeroOne() < 0.5f) {
```
Comment (isghj5) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:301]: wait... it calls ZeroOne twice and ignores the first one? (overwrites this later with _another one_?)

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 3]
```diff
@@ -5,6 +5,8 @@
  */
 
 #include "z_obj_ice_poly.h"
+#include "overlays/actors/ovl_Obj_Aqua/z_obj_aqua.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
 
 #define FLAGS 0x00000010
```
Comment (isghj5) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:11]:
```suggestion
#define FLAGS ACTOR_FLAG_10
```

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 4]
```diff
@@ -32,37 +34,327 @@ const ActorInit Obj_Ice_Poly_InitVars = {
...
+    thisx->params &= 0xFF;
+    thisx->uncullZoneForward = 5600.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Ice_Poly/ObjIcePoly_Destroy.s")
+    Actor_SetScale(thisx, thisx->params * 0.01f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Ice_Poly/func_80931828.s")
+    for (i = 0; i < ARRAY_COUNT(this->colliders1); i++) {
+        Collider_InitAndSetCylinder(globalCtx, &this->colliders1[i], thisx, &sCylinderInit1);
+        Collider_InitAndSetCylinder(globalCtx, &this->colliders2[i], thisx, &sCylinderInit2);
```
Comment (isghj5) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:92]: these have to be thisx to match? interesting

**Review by isghj5 — APPROVED** [review]
Looks good to me, nothing but minor thoughts

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:301]: It's not ignored, it's used in the expression below.

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c (hunk 4 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:92]: I don't think so, but thisx was needed in the function for the collider assignments, and from what I know, if you need to use thisx, you need to use it as much as possible. I can change those 2 back, but that wasn't what I heard to do.

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c (hunk 2 above)
Comment (isghj5) [reply src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:301]: oh I visually skipped over it, focusing on the re-use nvm

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 5]
```diff
@@ -32,37 +34,327 @@ const ActorInit Obj_Ice_Poly_InitVars = {
...
+          ((this->colliders2[0].base.ac->id != ACTOR_OBJ_AQUA) &&
+           (this->colliders2[0].info.acHitInfo->toucher.dmgFlags == 0x800)) ||
+          ((this->colliders2[0].base.ac->id == ACTOR_OBJ_AQUA) &&
+           (this->colliders2[0].base.ac->params == OBJAQUA_1)))) ||
+        ((this->colliders2[1].base.acFlags & AC_HIT) &&
+         ((this->colliders2[1].base.ac == NULL) ||
+          ((this->colliders2[1].base.ac->id != ACTOR_OBJ_AQUA) &&
+           (this->colliders2[1].info.acHitInfo->toucher.dmgFlags == 0x800)) ||
+          ((this->colliders2[1].base.ac->id == ACTOR_OBJ_AQUA) &&
+           (this->colliders2[1].base.ac->params == OBJAQUA_1))))) {
```
[the remark this answers is not in the document]
Comment (AngheloAlf) [reply src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:202]: oh wtf is this

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Good work on this my friend.

**Pull request merged**